crates/factor-outbound-http/src/lib.rs

```diff
@@ -1,5 +1,7 @@
 mod spin;
 mod wasi;
+mod wasi_2023_10_18;
+mod wasi_2023_11_10;
```

Comment:
Why do we need these in this crate as opposed to using the definitions in `spin_world`?

Reply:
These adapters need customized bindgen options to be compatible with the base `wasmtime-wasi{,-http}` impls.

(The _real_ answer is that I just blindly adapted @alexcrichton's code from `spin-core`. :sweat_smile:)